### ISSUE TITLE
[Bug]: Very slow and non usable using Windows 11

### ISSUE THREAD

**Issue by dhaeb** [BelledonneCommunications/linphone-desktop#688]
### Context

**For which purpose do you use the project ?**
Local VOIP on my computer
**With which software/hardware it is integrated ?**

**Did you use sip.linphone.org or a different SIP service (in this case specify which one and which version) ?**
I use _sipgate.de_

### General information

- Device: 
Device name	<anonymized>
Processor	11th Gen Intel(R) Core(TM) i7-1165G7 @ 2.80GHz   2.80 GHz
Installed RAM	32,0 GB (31,3 GB usable)
Device ID	51754942-F14C-4BCA-9DB5-D4B95EB5895A
Product ID	00330-53730-87233-AAOEM
System type	64-bit operating system, x64-based processor
Pen and touch	Pen and touch support with 10 touch points

- OS:
Edition	Windows 11 Pro
Version	21H2
Installed on	‎21.‎10.‎2021
OS build	22000.795
Serial number	5CG1264SSR
Experience	Windows Feature Experience Pack 1000.22000.795.0

- Version of the App: 4.4.8
- Version of the SDK: ?
-

### Expected behaviour

I wanted to do a simple call with the Windows client calling a number. However, the desktop app becomes non-responsive:
![image](https://user-images.githubusercontent.com/3354824/183420557-554b8e04-16a7-4c72-ab73-87a817b541f2.png)
This is already happening as soon as you open the app, but itensifies over time. 

### To Reproduce

Everything is just laggy as soon as I open the app. It is responding slowly, like in a timelapse. Feels like software running on constrained resource conditions. I guess we have a severe memory leak or something here... it starts normalish but gots so much worse using it over time...

**Comment by 0xabu** [comment]
Same issue here. It doesn't appear to be a memory leak, just something burning a core (busy-waiting?) for lengthy multi-second delays before servicing the UI. I don't recall seeing anything like this on previous versions.

**Comment by 0xabu** [comment]
I tried some prior versions:
4.4.8 - repro
4.4.7 - repro
4.4.0 - repro
4.3.2 - no repro

**Comment by ekutner** [comment]
Same here

**Comment by Aarkon** [comment]
Confirming the issue on Windows 10. Been persistent since the last three or so versions.

**Comment by MasterPMax** [comment]
I can confirming the issue on Windows 10 (21H2) to. It works with the last 32 bit version (4.3.2).

**Comment by me1299** [comment]
Just this started for me this morning as well. It's unusable now. High CPU usage and Linphone is not responding according to the task manager.

**Comment by dspinellis** [comment]
I saw the slow (1-4" lag for each mouse click and 100% core load) behavior as soon as the program starts up.  It appears in versions 5.0.3, 5.0.4, 4.4.11, and not in version 4.3.2.

**Comment by yossigrosskopf** [comment]
Same here, freezes on launch after a clean installation, cant even pass the Assistant setup

**Comment by TheGreyDiamond** [comment]
I have the same issue in Windows 10.

**Comment by jadaml** [comment]
I have Linphone installed on two machines, one with Win10, and the other with Win11
Win11: works flawlessly, though did not used it for long period of time just yet (closed it within a minute)
Win10: Starts slow, then stops working, then returns for a while, then stops again, then returns; this repeats. Wasn't able to configure it yet either. Did not noticed whether each stopping lasts for longer than the previous one or not, but I've always lost my patience when tried to configure it.

Win10
```
Processor:   Intel(R) Core(TM) i7-4820K CPU @ 3.70GHz   3.70 GHz
Memory Size: 16.0 GB
System Type: 64-bit
```

Win11
```
Processor	Intel(R) Core(TM) i7-1065G7 CPU @ 1.30GHz   1.50 GHz
Installed RAM	16.0 GB (15.7 GB usable)
System type	64-bit operating system, x64-based processor
```

**Comment by ramonchiara** [comment]
Same issue here... Can't even drag the window.

**Comment by audacity76** [comment]
Happens to me as well. :(

```
Processor: 12th Gen Intel(R) Core(TM) i7-12700T   1.40 GHz
RAM: 64,0 GB
OS Build: 22621.1105
```

**Comment by audacity76** [comment]
Just found out that if I close Google Drive (Sync) App performance is back again!

**Comment by ckirkham01** [comment]
Having the same issue on Windows 11 when running as a non-admin user.  When I run as an admin user they issue is not there (permission issue?).  

Looking at Process Explorer I see LOTS of cycles on linphone.exe!PluginDataAPI.  

![image](https://user-images.githubusercontent.com/104623204/215745944-851d0219-24d1-4cb8-864a-f5a0486dc79b.png)

**Comment by Nico-de-Vries** [comment]
Installed Linphone on Windows 11. Almost frozen UIF. Found this tread. Uninstalled.

**Comment by Jade64** [comment]
> Just found out that if I close Google Drive (Sync) App performance is back again!

Same here, Windows 11 and several Linphone versions. Up to newest 5.0.10.
Workaround with closing Google Drive App worked for me as well!!

**Comment by julonexus** [comment]
What is the link between the Google Drive App and Linphone... 😐

**Comment by b1scoito** [comment]
Terminating the Google Drive process worked for me too.

**Comment by ckirkham01** [comment]
Must be a different issue for me - don't have Google Drive App running.  I do have OneDrive tried shutting that down and it did not make a difference - the lag is still there

**Comment by jhansen858** [comment]
Having the same problem on quite a few computers.  Hope a solution can be found soon.  On affected computers the app is completely unusable.

**Comment by b1scoito** [comment]
As https://github.com/BelledonneCommunications/linphone-desktop/issues/688#issuecomment-1410180053 mentioned, I'm getting abnormal high CPU cycle counts on this thread: 
![image](https://user-images.githubusercontent.com/17802984/219488804-1587d39f-b68f-447a-8a53-810c624f6e41.png)
Which passes by a lot of Qt5 functions. I haven't been able to find the exact cause of the lag caused by the Google Drive app.

**Comment by k-tahiro** [comment]
Same issue here without Google Drive process.

**Comment by julonexus** [comment]
The Linphone SDK is running on the main thread and the frequency of iteration is 20ms. So if something slow down the iteration, the application will freeze. 
The PluginDataAPI seems to be a false flag : I don't know why the stack from the Process Explorer is different from a local build but what is count in CPU cycle is the iterate from the exec of the main (which call Qt and Linphone).
You should check your logs to see what is wrong (network timeout? bad server configuration?)

**Comment by agemuend** [comment]
I'm also hit by this using Windows 10 and no Google Drive. 

I don't see anything special in the logs, apart from this: 
2023-04-21 12:22:40:498 [/mediastreamer] MESSAGE Fail to load plugin C:\Program Files\Linphone\plugins\mediastreamer\libmsopenh264.dll with altered search path: error 126
2023-04-21 12:22:40:498 [/mediastreamer] ERROR Fail to load plugin C:\Program Files\Linphone\plugins\mediastreamer\libmsopenh264.dll: error 126

**Comment by Bakobiibizo** [comment]
Same issue, windows 10, it was working previously and now it is sooo laggy. Even tried to run in a VM on a fresh install of ubuntu and still no difference.

**Comment by julonexus** [comment]
> You should check your logs to see what is wrong (network timeout? bad server configuration? Accessing files? etc.)

Also, you may try to deactivate the logs. If it resolves the issue, I suggest to move your logs folder elsewhere.

**Comment by master-fuggi** [comment]
Same issue with Windows 11 and Linphone 5.0.15 - CPU load as high as 25 % and lag times of more than 5 seconds upon each mouse click without any registered SIP account and logging being disabled.

**Comment by nopeee535** [comment]
Are there any news ?
we have the same issue on a Dell XPS - Microsoft Windows 11 Pro - 22H2/22621 (without google drive) - LinPhone 5.0.18

**Comment by jhansen858** [comment]
We have some evidence that its related to a timeout scenario when trying to
write to disk but some invalid / unavailable path exists in the system.

**Comment by IVRNetworks** [comment]
Same issue, put in a separate ticket

**Comment by rafaelurben** [comment]
Can reproduce - this makes the app almost unusable.

Closing Google Drive does not fix the issue for me. (Windows 11 Pro 22H2/22621.2134)

**Comment by agemuend** [comment]
@julonexus I don't think its related to Google Drive or file I/O. It rather seems to be related to QtQuick Accessibility. 
I did a profiling run with Intel Vtune sampling drivers and it spends 60% of CPU time on QGuiApplicationPrivate::processMouseEvent, particularly in QQuickWindowPrivate::deliverHoverEvent and ::deliverPointerEvent. The call stack is huge, so I didn't go through it completely, but it seems to constantly call QQuickGroupBox::accessibilityActiveChanged and QQuickToolTip::accessibilityActiveChanged. So I'd guess that there is some misconfigured hover event in the QtQuick controls.  You can also see that the windows accesibility service (if its enabled) spikes with lots of CPU while Linphone runs. 
I tested using the latest linphone release 5.1.0-win64.

**Comment by julonexus** [comment]
Great findings. That could explains why the UI freeze when we activate the cursor decoration in Windows accessibility.
Also, that bug has been reported to Qt but there are nothing new about it.
https://bugreports.qt.io/browse/QTBUG-46865
Some fix has been done at https://bugreports.qt.io/browse/QTBUG-95114 but on 5.15.10. As we don't build Qt, we are using the 5.15.2 from the public repository for MSVC.
This framework is build with accessibility, so we cannot deactivate the feature from Qt.

Somebody tried to deactivate accessibility but without success : https://forum.qt.io/topic/132927/help-completely-disabling-accessibility

**Comment by agemuend** [comment]
QTBUG-95114 or rather QTBUG-97103 seems fitting indeed, I can see in the stacktrace that it traverses all childItems and calls setName() as well.

Do you know what the last Linphone version was that was built with a Qt < 5.15.1? Because they identified a commit from 5.15.1 as culprit. I could check if the older version shows the same behaviour or not. 

This user has found another workaround, implementing empty handlers for accessibility. I don't know if you would consider this: https://stackoverflow.com/questions/75082068/qt-qml-windows-app-very-slow-when-grammarly-app-is-running
I don't have a working build environment for Linphone on Windows right now (I usually develop under Linux), but if you can't reproduce it, I'd be willing to set one up. 

Unfortunately Linphone is completely unusable on affected systems right now.

**Comment by julonexus** [comment]
Thanks to repointing this post.
I saw it but as I get no documentations about QAccessible::installUpdateHandler and QAccessible::installRootObjectHandler on 5.15 but only on Qt4, I guessed that this workaround was about Qt4µ. I didn't read the important part "BTW: We are using Qt 5.15.2 opensource."  😅

I tried to build the application with 5.15.2 with this overload handlers and it seems to work. So good news :)
Also, I get some feedback on 5.15.14 that Accessibility works as expected.

The fix will be available for 5.1.1. I'll close the ticket when the application goes out.
Thanks you very much for your help.

**Comment by agemuend** [comment]
Fantastic news! Thanks!

**Comment by Bakobiibizo** [comment]
Started again with the last updated

**Comment by agemuend** [comment]
The fix has been commited yesterday, but it's not released yet. It will be included in 5.1.1.

**Comment by julonexus** [comment]
done ;)
